core/task-checks.go

```diff
@@ -49,32 +54,35 @@ func GetCommonChecks() *CommonChecks {
 	}
 
 	// Battery
-	cmd = exec.Command("acpi")
+	cmd = exec.Command("ls", powerPath)
 	output, err := cmd.CombinedOutput()
 	if err == nil {
-		if strings.Contains(string(output), "Discharging") {
-			cChecks.Battery = true
-		}
-		if strings.Contains(string(output), "Charging") || strings.Contains(string(output), "Not charging") {
-			cChecks.Battery = false
-		}
-		percentage := ""
-		for _, char := range string(output) {
-			if char >= '0' && char <= '9' {
-				percentage += string(char)
-			}
-			if char == '%' {
-				break
-			}
-		}
-		if percentage != "" {
-			perc, err := strconv.Atoi(percentage)
-			if err == nil {
-				if perc <= 30 {
-					cChecks.LowBattery = true
+		devices := strings.Split(string(output), "\n")
+		for _, dev := range devices {
+			if strings.Contains(dev, "BAT") {
+				devPath := path.Join(powerPath, dev)
+				statusPath := path.Join(devPath, "status")
```

Comment:
I'm considering just making all of these constants above rather than doing the path.Join here, it'll save a few microseconds. Any preferences?

Reply:
I agree with this approach